packages/core/src/render3/component_ref.ts

```
@@ -168,11 +168,17 @@ export class ComponentFactory<T> extends AbstractComponentFactory<T> {
     if (ngDevMode && (typeof ngJitMode === 'undefined' || ngJitMode) &&
         this.componentDef.debugInfo?.forbidOrphanRendering) {
       if (depsTracker.isOrphanComponent(this.componentType)) {
-        throw new RuntimeError(
+        const err = new RuntimeError(
             RuntimeErrorCode.RUNTIME_DEPS_ORPHAN_COMPONENT,
             `Orphan component found! Trying to render the component ${
                 debugStringifyTypeForError(
                     this.componentType)} without first loading the NgModule that declares it. It is recommended to make this component standalone in order to avoid this error. If this is not possible now, import the component's NgModule in the appropriate NgModule, or the standalone component in which you are trying to render this component. If this is a lazy import, load the NgModule lazily as well and use its module injector.`);
+
+        // In some settings the thrown error is swallowed, so we log it manually here to make sure
+        // it shows up in the console and/or logs.
+        console.error(`Rendering component failed with the error: ${err.message}`);
```

Wouldn't this cause it to show up twice if it doesn't get swallowed?

---

Yes it is!

Ideally we have to fix those test infra that discard the exceptions. But or now we want to enable the orphan component check cross g3, and we want to ensure that any breakage due to this exception shows up the message in some log file, otherwise people will be confused as to why their app is broken.

---

Couldn't we keep just the `console.error` in that case? I'm a bit afraid that it'll get too spammy otherwise.

---

I did some experiment and here is what I found: if the runtime error is not caught, then it breaks the test and the error message shows nicely as the first thing in the test log: [example](http://sponge2/4d64f4c8-c367-4d83-b470-83083c68bc1f) - Now assume we just `console.error` the error message and break the test by throwing a generic error (or we just let the pipeline to continue), then the runtime error will not show in the test log: [example](https://fusion2.corp.google.com/invocations/d99202f3-f18e-4f5d-9ffe-b842c45a91c5/targets/%2F%2Fcorp%2Ffinance%2Fworkflow%2Ffrontend:ts_tests_chrome-linux/tests)(to see the orphan component error you will need to go to the target log, which is a huge log, and search for it!)

Also to add, the situation where this runtime exception is discarded is rather rare (< 3% ) - so perhaps throwing the exception is still a good solution, but yeah at the cost of duplication. I'll try to fix those tests later and see why they discard such important errors